<launch>
    <param name ="/use_sim_time" value="true"/>
    <!--sim_planner1-->
<!--Input######################################################################################################-->
    <node pkg="rosbag" type="play" name="rosbag" args="$(find ccmapping)/bag/sim_planner4.bag -r2 --clock"
        launch-prefix="bash -c 'sleep 2; $0 $@' ">
    </node>

<!--rviz#######################################################################################################-->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find ccmapping)/launch/config/flvis_ccmapping.rviz" />

<!--ccmapping##################################################################################################-->
    <arg name="node_start_delay"  default="1.0" />
    <param name="/ccmapping_configfile" type="string" value="$(find ccmapping)/launch/config/d435i_ccmapping_config.yaml"/>

    <!-- Manager -->
    <node pkg="nodelet" type="nodelet"
          name="ccmapping_nodelet_manager" args="manager" output="screen">
          <param name="num_worker_threads" value="2" />
    </node>

    <node pkg="nodelet" type="nodelet" args="load ccmapping/LocalMapNodeletClass ccmapping_nodelet_manager"
        name="ccmapping_LocalMapNodeletClass_loader" output="screen"
        launch-prefix="bash -c 'sleep 2; $0 $@' ">
        <remap from="/ccmapping/pc"          to="/camera/depth/color/points"/>
        <remap from="/ccmapping/pose"        to="/mavros/local_position/pose"/>
    </node>

    <node pkg="nodelet" type="nodelet" args="load ccmapping/GlobalMapNodeletClass ccmapping_nodelet_manager"
        name="ccmapping_GlobalMapNodeletClass_loader" output="screen"
        launch-prefix="bash -c 'sleep 1; $0 $@' ">
    </node>

</launch>







